libraries/type-length-value/src/state.rs

```diff
@@ -85,28 +85,34 @@ fn get_discriminators_and_end_index(
     let mut discriminators = vec![];
     let mut start_index = 0;
     while start_index < tlv_data.len() {
-        let (type_start, length_start, value_start) = get_indices_unchecked(start_index);
-        if tlv_data.len() < length_start {
+        let tlv_indices = get_indices_unchecked(start_index, 0);
```

Comment:
I see what you mean here... up to you, but you could add a comment here to explain the weirdness on `0` and why it's there

Reply:
Yep, added a comment